mkosi/__init__.py

```diff
@@ -4718,8 +4718,12 @@ def _link_output(
 
 
 def link_output(args: MkosiArgs, root: Path, artifact: Optional[BinaryIO]) -> None:
+
     with complete_step("Linking image file…", f"Linked {path_relative_to_cwd(args.output)}"):
         if args.output_format in (OutputFormat.directory, OutputFormat.subvolume):
+            if not root.exists():
+                return
```

Comment:
Please also turn the `assert artifact` line below into an if check that skips if `artifact` is `None`

Reply:
`assert artifact is None` on line 4726 is correct. There should be no artifact but there should be a directory or subvolume to make read-only and copy to the output directory as appropriate. Unless I misunderstood, I'd end up with:

```python
if not root.exists() or artifact is None:
    return
```

I may have simply misunderstood.

Reply:
It's not about line 4726, it's about the `assert artficact` line further down